src/test/java/edu/usf/cutr/gtfsrtvalidator/test/feeds/TripUpdateFeedTest.java

```diff
@@ -0,0 +1,109 @@
+
+import edu.usf.cutr.gtfsrtvalidator.FeedMessageTest;
+import edu.usf.cutr.gtfsrtvalidator.validation.entity.CheckTripId;
+import edu.usf.cutr.gtfsrtvalidator.validation.entity.StopTimeSequanceValidator;
+import edu.usf.cutr.gtfsrtvalidator.validation.entity.VehicleIdValidator;
+import edu.usf.cutr.gtfsrtvalidator.validation.gtfs.StopLocationTypeValidator;
+import org.junit.Test;
+
+
+public class TripUpdateFeedTest extends FeedMessageTest{
+    
+    public TripUpdateFeedTest() throws Exception{}
+    
+    @Test
+    public void testTripIdValidation(){
+        
+        CheckTripId tripIdValidator = new CheckTripId();
+        
+        // feedMessage already have a valid trip_id set in InitRequiredFeedFields() method
+        errors = tripIdValidator.validate(gtfsData, feedMessage.build());
+        assertEquals(0, errors.getOccurrenceList().size());
+        
+        tripDescriptor.setTripId("100");
+        tripUpdate.setTrip(tripDescriptor);
+        feedEntity.setTripUpdate(tripUpdate);
+        feedMessage.setEntity(0, feedEntity);
+        
+        errors = tripIdValidator.validate(gtfsData, feedMessage.build());
+        assertEquals(1, errors.getOccurrenceList().size());
+        
+        clearAndInitRequiredFeedFields();
+    }
+    
+    @Test
+    public void testStopSequenceValidation() {
+        StopTimeSequanceValidator stopSequenceValidator = new StopTimeSequanceValidator();
+        
+        //ordered stop sequence 1, 5
+        stopTimeUpdate.setStopSequence(1);
+        tripUpdate.addStopTimeUpdate(stopTimeUpdate);
+        stopTimeUpdate.setStopSequence(5);
+        tripUpdate.addStopTimeUpdate(stopTimeUpdate);
+        feedEntity.setTripUpdate(tripUpdate);
+        feedMessage.setEntity(0, feedEntity);
+        
+        errors = stopSequenceValidator.validate(gtfsData, feedMessage.build());
+        assertNull(errors);
+        
+        /* Adding stop sequence 3. So, the stop sequence now is 1, 5, 3 which is unordered.
+        So, the validation fails and the assertion test passes
+        */
+        stopTimeUpdate.setStopSequence(3);
+        tripUpdate.addStopTimeUpdate(stopTimeUpdate);
+        feedEntity.setTripUpdate(tripUpdate);
+        feedMessage.setEntity(0, feedEntity);
+        
+        errors = stopSequenceValidator.validate(gtfsData, feedMessage.build());
+        assertEquals(1, errors.getOccurrenceList().size());
+        
+        clearAndInitRequiredFeedFields();
+    }
+    
+    @Test
+    public void testVehicleIdValidation() {
+        VehicleIdValidator vehicleIdValidator = new VehicleIdValidator();
+        
+        vehicleDescriptor.setId("1");
+        tripUpdate.setVehicle(vehicleDescriptor);
+        feedEntity.setTripUpdate(tripUpdate);
+        feedMessage.setEntity(0, feedEntity);
+        
+        errors = vehicleIdValidator.validate(gtfsData, feedMessage.build());
+        assertNull(errors);
+        
+        //unable to set Id = null as it's throwing NullPointerException.
```

Comment:
Can you try instantiating the `vehicleDesriptor` but not setting the ID?  In that case I'm assuming it will still be null.